internal/services/storage/storage_management_policy_resource.go

```diff
@@ -285,7 +287,21 @@ func resourceStorageManagementPolicyCreateOrUpdate(d *pluginsdk.ResourceData, me
 	// The name of the Storage Account Management Policy. It should always be 'default' (from https://docs.microsoft.com/en-us/rest/api/storagerp/managementpolicies/createorupdate)
 	mgmtPolicyId := parse.NewStorageAccountManagementPolicyID(rid.SubscriptionId, rid.ResourceGroupName, rid.StorageAccountName, "default")
 
-	// TODO: support Requires Import
+	if d.IsNewResource() {
+		// This lock is to protect the existence checking when two storage mgmt policies are being created at the same time.
+		locks.ByID(mgmtPolicyId.ID())
```

Comment:
I'm not sure we should go this route quite yet. We should do a separate PR with it's own test to confirm that this will fix that issue.

Reply:
Yeah, I've removed the locks.